account_analytic_default_account/models/account_analytic_default_account.py

```diff
@@ -0,0 +1,148 @@
+# -*- coding: utf-8 -*-
+# Copyright 2018 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
+
+from odoo import api, fields, models
+
+
+class AccountAnalyticDefaultAccount(models.Model):
+    _inherit = "account.analytic.default"
+
+    account_id = fields.Many2one(
+        'account.account', string='Account',
+        ondelete='cascade',
+        help="Select the account corresponding to an analytic account "
+             "which will be used on the lines of invoices or account moves"
+    )
+
+    account_get_domain_keys = [
+        'product_id', 'partner_id', 'user_id',
+        'date', 'company_id', 'account_id',
+    ]
+
+    def _account_get_domain(self, **kw):
+        """Build account.analytic.default domain.
+
+        :param kw: filter keys.
+            Available filter keys are defined into `account_get_domain_keys`.
+        :return: a search domain with all required keys' leaves.
+            Each key will be searched for both False and specified value.
+            For instance: [
+                ('product_id'), '=', False), '|', [('product_id'), '=', 100)
+            ]
+        """
+        domain = []
+        account_get_domain_keys = self.account_get_domain_keys[:]
```

Comment:
you don't need this anymore as the keys are controlled by the main method signature
```
def _account_get_domain(self, **kw):
    """Build account.analytic.default domain.
    :param kw: filter keys.
        Available filter keys are defined into `account_get_domain_keys`.
    :return: a search domain with all required keys' leaves.
        Each key will be searched for both False and specified value.
        For instance: [
            ('product_id'), '=', False), '|', [('product_id'), '=', 100)
        ]
    """
    domain = []
    date_val = kw.pop('date', None)
    for key, value in kw.items():
        if value is not None:
            domain += ['|', (key, '=', value)]
        domain += [(key, '=', False)]

    if date_val:
        domain += ['|', ('date_start', '<=', date_val),
                   ('date_start', '=', False)]
        domain += ['|', ('date_stop', '>=', date_val),
                   ('date_stop', '=', False)]
    return domain
```